fc/parsing/actions.py

```diff
@@ -910,6 +913,29 @@ def modify_model(self, model, units):
         self._purge_unused_mathematics(output_variables)
         # TODO: Any final consistency checks on the model?
 
+    def _sanity_check(self):
+        """Perform initial sanity checks on the protocol interface."""
+
+        # Variables can only appear as input or output once
+        # If a variable appears as an input and an ouput, both must have the same units
+        input_units = {}
+        output_units = {}
+        for var in self.inputs:
+            if var.rdf_term in input_units:
+                raise ProtocolError('The variable ' + str(var.rdf_term) + ' was specified as an input twice.')
+            input_units[var.rdf_term] = var.units
+        for var in self.outputs:
+            if var.rdf_term in output_units:
+                raise ProtocolError('The variable ' + str(var.rdf_term) + ' was specified as an output twice.')
+            output_units[var.rdf_term] = var.units
+            units = input_units.get(var.rdf_term)
+            if var.units is not None and units is not None and var.units != units:
+                raise ProtocolError(
+                    'The variable ' + str(var.rdf_term) + ' appears as input and output, but with different units.')
+
+        # TODO Check defines
+        # TODO Check clamps
```

Comment:
Clamps could be done now. Defines aren't implemented yet (but will be `DeclareVariable`).